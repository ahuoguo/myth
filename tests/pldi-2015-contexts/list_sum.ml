#use "context.decls"

let rec fold (l:list) (f:nat -> nat -> nat) (acc:nat) : nat =
  match l with
  | Nil -> acc
  | Cons (x, l) -> fold l f (f acc x)
;;

let list_sum : list -> nat |>
  { [] => 0
  | [1] => 1
  | [2; 1] => 3
  } = ?
