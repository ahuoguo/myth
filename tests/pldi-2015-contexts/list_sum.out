(* Generated Thu Nov 13 04:28:00 2014 *)
let list_sum : list -> nat =
  fun (l1:list) -> fold l1 plus O
;;
